### DISCUSSION TITLE
switch thermosatt to power shedding mode only after X sec

### DISCUSSION THREAD

**Discussion by KipK** [jmcollin78/versatile_thermostat#672] [Ideas]
Circuit braker like Linky only thrills after 30-60sec giving some time for the power shedding to activate in the house. 
Could a parameter be added to only starts power shedding mode only if power exceed capacity after X seconds?

I have an automation to adapt my EV charging rate. But Versatile receive the same power data as the EVSE, and it put the thermostat on power saving mode instantly. The EV charge will takes like 2-3 seconds to adapt its current to capacity. Then the thermostat starts again after this time. 
Having the ability to set this delay will prevent those useless cutoff.

**Comment by jmcollin78** [comment]
Hello @KipK,

I think this is not so simple. VTherm do the shedding when the current power + the max consumption of a VTherm that should be turned on is exceeded. So the shedding is not activated when power consumption is exceeded but when it could be if the heater is turned of. So because of all the delays in power consumption update in HA, this is really not precise. 
So waiting for 2 or 3 secondes will not change anything (ev charger will update but will HA be updated also ? What about the others potentially Vtherm that have starts 5 or 10 sec before but consumption is not reported).

My proposal, is to accept that sometimes (this should be rare), the VTherm turns into power shedding. This is never long because the conditions are re-evaluated at each cycle. That is exaclty what I does (but with Solar Optimizer). Depending of the cycles of each devices, some VTherm could go to power shedding but that is not an issue.

I hope this helps.

**Reply by KipK** [reply]
I get it, I can live with that :)
